### PR TITLE
Adapting to recent HPX build system changes

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

**Comment by msimberg** [comment]
@hkaiser what got rid of the linker problems in the end?

**Comment by msimberg** [comment]
@hkaiser what got rid of the linker problems in the end?

File: cmake/Phylanx_SetupTarget.cmake [hunk 1]
```diff
@@ -99,6 +104,21 @@ function(phylanx_setup_target target)
       string(REPLACE ";" ":" _prefix "${_prefix}")
     endif()
 
+    if(NOT target_STATIC_LINKING)
+      if(HPX_WITH_DYNAMIC_HPX_MAIN AND ("${_type}" STREQUAL "EXECUTABLE"))
+        if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+          phylanx_info("phylanx_libs: " ${phylanx_libs})
+          set_target_properties(${target} PROPERTIES LINK_FLAGS "-Wl,-wrap=main")
```
Comment (msimberg) [inline cmake/Phylanx_SetupTarget.cmake]: Do you really want to be setting these linker flags here? I know you're following HPX master pretty closely but this feels like unnecessarily using implementation details when we have alternatives (`hpx_setup_target`/`add_hpx_library`, and hopefully `HPX_LIBRARIES` back in action soon again) until we know that `target_link_libraries` works correctly.

Comment (hkaiser) [reply cmake/Phylanx_SetupTarget.cmake]: Those have no effect anyways (for whatever reason). Things now link but still fail at runtime (e.g. https://circleci.com/gh/STEllAR-GROUP/phylanx/32106).

**Comment by NK-Nikunj** [comment]
Here are the changes I made:

- [x] Remove Phylanx_SetupHPX in favor of `find_package(HPX REQUIRED)`
- [x] Comment out all HPX related flags in Phylanx_SetupTarget and call hpx_setup_target with the target type. This allows hpx_setup_target to add any hpx related flags to the target.

**Comment by hkaiser** [comment]
@NK-Nikunj I had to apply some changes to your patch to retain certain Phylanx specific build settings. Please re-check.

**Comment by NK-Nikunj** [comment]
@hkaiser LGTM!

**Comment by aurianer** [comment]
Thanks @NK-Nikunj  & @hkaiser !

**Comment by hkaiser** [comment]
This is superseded by #1063.

**Pull request merged**